.github/workflows/kind_e2e.yaml

```diff
@@ -0,0 +1,140 @@
+# ------------------------------------------------------------
+# Copyright (c) Microsoft Corporation and Dapr Contributors.
+# Licensed under the MIT License.
+# ------------------------------------------------------------
+name: E2E tests on KinD
```

Comment:
This should be Run E2E tests ?
Kind is just one platform that is being tested?

Reply:
I'll rename to include "Self-Hosted"